app/coffee/DocArchiveManager.coffee

```diff
@@ -79,11 +79,43 @@ module.exports = DocArchive =
 				MongoManager.upsertIntoDocCollection project_id, doc_id.toString(), mongo_doc, (err) ->
 					return callback(err) if err?
 					logger.log project_id: project_id, doc_id: doc_id, "deleting doc from s3"
-					request.del options, (err, res, body)->
-						if err? || res.statusCode != 204
-							logger.err err:err, res:res, project_id:project_id, doc_id:doc_id, "something went wrong deleting doc from aws"
-							return callback new Errors.NotFoundError("Error in S3 request")
-						callback()
+					DocArchive._deleteDocFromS3 project_id, doc_id, callback
+
+	destroyAllDocs: (project_id, callback = (err) ->) ->
+		MongoManager.getProjectsDocs project_id, {include_deleted: true}, {_id: 1}, (err, docs) ->
```

Comment:
Is there any locking in place that prevents new inserts after this call?

I am thinking about a shared project in which the owner triggers the deletion process, while a collaborator keeps inserting new changes.

Reply:
Projects are soft-deleted, so this shouldn't be being called at a time when the project is active. However, with regard to the methods in docstore it is the caller's responsibility to ensure that the project has been flushed first. - As this is an internal API this is under the control of the framework, so it's reasonable not to have any specific locking for this here.

Reply:
Good question! By the time this code is called, the project will already have been archived and soft-deleted, so there won't be any editing activity going on.

Reply:
Alright, thanks for the clarification.